Calculator/Calculator/Model/Formula.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  Formula.swift
+//  Calculator
+//
+//  Created by 비모 on 2023/06/05.
+//
+
+struct Formula {
+    var operands: CalculatorItemQueue<Double>
+    var operators: CalculatorItemQueue<Operator>
+    
+    mutating func result() -> Double {
+        var result: Double = 0.0
+        
+        if operands.isEmpty {
+            return result
+        }
+        
+        if operands.count == 1, let number = operands.dequeue() {
+            return number
+        }
+        
+        guard let lhs = operands.dequeue() else {
+             return result
+        }
+        
+        result = lhs
+        
+        while operands.peek() != nil {
+            guard let rhs = operands.dequeue() else {
+                return result
+            }
+            
+            guard let `operator` = operators.dequeue() else {
+                return result
+            }
+            
+            result = `operator`.calculate(lhs: result, rhs: rhs)
+        }
```

Comment:
optional binding은 while문으로도 가능합니다.
따라서 현재 코드는 아래처럼 표현할 수 있을 것 같아요.
```suggestion
        while let `operator` = operators.dequeue(),  let rhs = operands.dequeue() {
            result = `operator`.calculate(lhs: result, rhs: rhs)
        }
```

만약 operands의 dequeue 반환값이 nil이면 while문을 빠져나갈 것 같아요. 
때문에 `operands.peek() != nil`도 필요 없어질 것 같습니다.
비모 생각은 어떤가요?

Reply:
`while`문을 통해서 옵셔널 바인딩을 할 수 있군요! 적용하여 코드 수정하도록 하겠습니다. 감사합니다!